File: packages/dcflight/android/src/main/kotlin/com/dotcorr/dcflight/components/FrameworkComponentsReg.kt
/*
 * Copyright (c) Dotcorr Studio. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.dotcorr.dcflight.components

import android.util.Log
import com.dotcorr.dcflight.components.DCFFlutterWidgetComponent

/**
 * Registration class for DCFlight framework components
 */
object FrameworkComponentsReg {
    private const val TAG = "FrameworkComponentsReg"

    /**
     * Register framework-level components
     */
    @JvmStatic
    fun registerComponents() {
        // Register FlutterWidget component for embedding Flutter widgets
        DCFComponentRegistry.shared.registerComponent("FlutterWidget", DCFFlutterWidgetComponent::class.java)
        Log.d(TAG, "✅ Registered FlutterWidget component")
        
        // Register core framework components
        DCFComponentRegistry.shared.registerComponent("View", DCFViewComponent::class.java)
        Log.d(TAG, "✅ Registered View component")
        
        DCFComponentRegistry.shared.registerComponent("Text", DCFTextComponent::class.java)
        Log.d(TAG, "✅ Registered Text component")
        
        DCFComponentRegistry.shared.registerComponent("ScrollView", DCFScrollViewComponent::class.java)
        Log.d(TAG, "✅ Registered ScrollView component")
        
        DCFComponentRegistry.shared.registerComponent("ScrollContentView", DCFScrollContentViewComponent::class.java)
        Log.d(TAG, "✅ Registered ScrollContentView component")
        
        DCFComponentRegistry.shared.registerComponent("Viewport", DCFViewportComponent::class.java)
        Log.d(TAG, "✅ Registered Viewport component")
    }
}

File: packages/dcflight/ios/Classes/Coordination/Components/FrameworkComponentsReg.swift
/*
 * Copyright (c) Dotcorr Studio. and affiliates.
 *
 * Licensed under the PolyForm Noncommercial License 1.0.0.
 * Commercial use requires a license from DotCorr.
 */

import Flutter
import UIKit
import dcflight

@objc public class FrameworkComponentsReg: NSObject {
    @objc public static func registerWithRegistrar(_ registrar: FlutterPluginRegistrar) {
        registerComponents()
    }

    @objc public static func registerComponents() {
        // Register FlutterWidget component for embedding Flutter widgets
        DCFComponentRegistry.shared.registerComponent("FlutterWidget", componentClass: DCFFlutterWidgetComponent.self)
        
        // Register core framework components
        DCFComponentRegistry.shared.registerComponent("View", componentClass: DCFViewComponent.self)
        
        DCFComponentRegistry.shared.registerComponent("Text", componentClass: DCFTextComponent.self)

        DCFComponentRegistry.shared.registerComponent("ScrollView", componentClass: DCFScrollViewComponent.self)
     
        
        DCFComponentRegistry.shared.registerComponent("ScrollContentView", componentClass: DCFScrollContentViewComponent.self)
        
        DCFComponentRegistry.shared.registerComponent("Viewport", componentClass: DCFViewportComponent.self)
        
        // Register TouchableOpacity component (framework-level, used by Button)
        DCFComponentRegistry.shared.registerComponent("TouchableOpacity", componentClass: DCFTouchableOpacityComponent.self)
        
    }
}
